c7n/resources/iam.py

```diff
@@ -825,6 +826,51 @@ def process(self, resources, event=None):
             return [r for r in resources if self._num_inline_policies(c, r) > 0]
         return [r for r in resources if self._num_inline_policies(c, r) == 0]
 
+@User.filter_registry.register('all-user-policies')
+class AllUserPolicies(ValueFilter):
+    """Applies a filter which is given all policies in a hash as an input, suitable
+    for finding details on aggregates / counts of matching policies.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: iam-users-with-with-inline-policies
+            resource: iam-user
+            filters:
+              - type: all-user-policies
+                key: length(InlinePolicyNames)
+                value: 0
+                op: gt
+    """
+
+    schema = type_schema('all-user-policies', rinherit=ValueFilter.schema)
+    permissions = ('iam:ListAttachedUserPolicies', 'iam:ListUserPolicies')
+
+    def user_policies(self, user_set):
+        client = local_session(self.manager.session_factory).client('iam')
+        for u in user_set:
+            if 'c7n:AllUserPolicies' not in u:
+                u['c7n:AllUserPolicies'] = {}
+            mapping = u['c7n:AllUserPolicies']
```

Comment:
could be simplified to `mapping = u.get('c7n:AllUserPolicies', {})`

Reply:
This is very belated - but would that still set the value on u? We want it written back if it isn't there, since that keeps it on the resource